<?xml version="1.0"?>
<launch>
	
	<!-- <node name="heightmap_node_itst" pkg="velodyne_height_map" type="heightmap_node"> -->
	<!-- 	<param name="cell_size" value="0.2"/> -->
	<!-- 	<param name="full_clouds" value="true"/> -->
	<!-- 	<param name="grid_dimensions" value="500"/> -->
	<!-- 	<param name="height_threshold" value="0.15"/> -->
	<!-- 	 -->
	<!-- 	<remap from="/velodyne_points" to="/perfect_velodyne/normal" /> -->
	<!-- 	<remap from="/velodyne_obstacles" to="/velodyne_obstacles/itst" /> -->
	<!-- </node> -->

	<node name="rm_groundbynormal" pkg="pcl_tutorials" type="rm_groundbynormal">
		<param name="rm_normal" value="-0.9"/>
		
		<remap from="/velodyne_points" to="/perfect_velodyne/normal" />
		<remap from="/velodyne_obstacles" to="/velodyne_obstacles/itst" />
	</node>



	<node pkg="perfect_velodyne" type="normal_estimation_refine_smoothing" name="NormalEstimationForVelodyne">
		<remap from="/velodyne_points" to="/velodyne_points/slam" />

		<param name="HNN" value="3l" />
		<param name="MAX_RANGE" value="120.0f" />
		<param name="MIN_RANGE" value="0.5f" />
		<param name="VR" value="1.00f" />
		<param name="skip" value="1.0f" />
		<param name="vector_horizon" value="0.1f" />
		<param name="vector_vertical" value="1.5f" />
		<param name="DENS" value="0.5f" />
		<param name="CURV" value="0.1f" />
		<param name="DISP" value="1l" />

	</node>
	
	<node name="histogram_create_normal" pkg="itst_desc" type="histogram_create_normal" output="screen" >
		<rosparam file="$(find ref_data_manager)/config/itst_info.yaml"/>
	
		<remap from="/lcl_imu" to="/odometer" />
		<remap from="/velodyne_points" to="/velodyne_obstacles/itst" />
	</node>
</launch>

